### PR TITLE
WRN-3834: Migrate Input and internal/SharedStateDecorator unit tests to Testing Library. 

### PR DESCRIPTION
**Pull request by alexandrumorariu**
### Checklist
* [X] I have read and understand the contribution guide
* [ ] A CHANGELOG entry is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
Migrated tests from Enzyme to Testing Library.

### Resolution
Changed test structure to work with Testing Library.

### Links
WRN-3834

### Comments

### CODE REVIEW

**Review by adrian-cocoara-lgp — APPROVED** [review]
LGTM

File: Input/tests/Input-specs.js [hunk 1]
```diff
@@ -1,168 +1,154 @@
-import {mount} from 'enzyme';
+import '@testing-library/jest-dom';
+import {fireEvent, render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
 import Input from '../Input';
 import Spotlight from '@enact/spotlight';
```
Comment (stanca-pop-lgp) [inline Input/tests/Input-specs.js]: Please move this import to the top.

File: Input/tests/Input-specs.js [hunk 2]
```diff
@@ -1,168 +1,154 @@
...
 		const actual = handleChange.mock.calls[0][0].value;
 
-		expect(actual).toBe(expected);
+		expect(actual).toBe(value);
 	});
 
 	test('should blur input on enter if dismissOnEnter', () => {
-		const node = document.body.appendChild(document.createElement('div'));
 		const handleChange = jest.fn();
+		render(<Input onBlur={handleChange} dismissOnEnter />);
```
Comment (stanca-pop-lgp) [inline Input/tests/Input-specs.js]: Please sort props alphabetically.

**Review by stanca-pop-lgp — APPROVED** [review]
LGTM

**Pull request merged**